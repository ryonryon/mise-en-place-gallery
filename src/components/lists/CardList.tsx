import styled from "@emotion/styled";
import * as React from "react";

import SimpleCard from "../cards/BasicCard";

const StyledListDiv = styled.div`
  margin: 16px;
  height: 400px;
  overflow: scroll;
  width: 90%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;

  div {
    width: 43%;
  }
`;

interface Props {
  list: string[];
}

const CardList = ({ list }: Props) => (
  <StyledListDiv>
    {list.map((item: string, index: number) => (
      <SimpleCard key={index}>
        <h4>{item}</h4>
      </SimpleCard>
    ))}
  </StyledListDiv>
);

export default CardList;
