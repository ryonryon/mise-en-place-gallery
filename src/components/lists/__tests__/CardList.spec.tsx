import * as React from "react";
import * as renderer from "react-test-renderer";

import CardList from "../CardList";

describe("CardList test", () => {
  it("tests CardList test render correctly", () => {
    const cardList = renderer.create(<CardList list={["list item1", "list item2", "list item3"]} />).toJSON();

    expect(cardList).toMatchSnapshot();
  });
});
