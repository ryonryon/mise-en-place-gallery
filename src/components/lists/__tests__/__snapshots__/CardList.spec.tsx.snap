// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList test tests CardList test render correctly 1`] = `
<div
  className="css-1qosea9"
>
  <div
    className="css-1pcjs3q"
  >
    <h4>
      list item1
    </h4>
  </div>
  <div
    className="css-1pcjs3q"
  >
    <h4>
      list item2
    </h4>
  </div>
  <div
    className="css-1pcjs3q"
  >
    <h4>
      list item3
    </h4>
  </div>
</div>
`;
